tutorials/dapt-curation/code/utils.py

```diff
@@ -0,0 +1,389 @@
+    """
+    Redact PII on comment lines on a  given code dataset.
+
+    Args:
+        dataset (DocumentDataset): The dataset to be redacted.
+
+    Returns:
+        DocumentDataset: The redacted dataset.
+    """
+    # Extract relevant data from the Dask DataFrame
+    length = len(dataset)
+    data_dict = {
+        "documents": dataset.df.compute().to_dict(orient="records"),
+    }
+    # Serialize the dictionary to a JSON string
+    json_data = json.dumps(data_dict, indent=2)
+    redactd_json_data = comment_redaction(json_data)
+
+    # Deserialize the JSON string to a dictionary
+    redactd_data_dict = json.loads(redactd_json_data)
+
+    # Create a DataFrame from the dictionary
+    redactd_dataset_df = dd.from_pandas(
+        pd.DataFrame(redactd_data_dict["documents"]), npartitions=length
+    )
+
+    # Create a DocumentDataset instance
+    redactd_document_dataset = DocumentDataset(redactd_dataset_df)
+
+    return redactd_document_dataset
+
+
+def redact_pii(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Redacts personally identifiable information (PII) from a given dataset.
+
+    Args:
+        dataset (DocumentDataset): The dataset containing documents with PII.
+
+    Returns:
+        DocumentDataset: The redacted dataset with PII replaced by a generic value.
+    """
+    DEFAULT_MAX_DOC_SIZE = 3000000
+    redactor = Modify(
+        PiiModifier(
+            supported_entities=[
+                "PERSON",
+                "EMAIL_ADDRESS",
+                #                                "URL",
+                #                                "PHONE_NUMBER",
+                #                                "ADDRESS",
+                #                                "LOCATION"
+            ],
+            anonymize_action="replace",
+            #             batch_size=1000,
+            device="gpu",
+        ),
+        text_field="text",
+    )
+    return redactor(dataset)
+
+
+# -
+
+
+def dedupe(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Remove exact duplicates from the given DocumentDataset.
+
+    Args:
+        dataset (DocumentDataset): The dataset containing documents.
+
+    Returns:
+        DocumentDataset: The deduplicated dataset.
+    """
+    deduplicator = ExactDuplicates(id_field="id", text_field="text", hash_method="md5")
+    # Find the duplicates
+    duplicates = deduplicator(dataset)
+    docs_to_remove = duplicates.df.map_partitions(
+        lambda x: x[x._hashes.duplicated(keep="first")]
+    )
+    # Remove the duplicates using their IDs.
+    duplicate_ids = list(docs_to_remove.compute().id)
+    dataset_df = dataset.df
+    deduped = dataset_df[~dataset_df.id.isin(duplicate_ids)]
+    return DocumentDataset(deduped)
+
+
+# +
+def filter_txt_lines(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Discard text files based on lines.
+    """
+    dataset_df = dataset.df
+    #     print(len(dataset_df.index))
```

Comment:
Remove?
Since there appears to be multiple occurances, I won't comment further on this, please remove all debugging code from the PR for cleaning purposes.

Reply:
done